Authentication support

iperevoschikov opened the pull request:
Можно смотреть только последний коммит

Commit:
Squashed 'cassandra-local/' changes from 49cd0e5..3c224fd

3c224fd Version 1.4.x
1b68720 Merge pull request #2 from skbkontur/authentication
6375a97 Add ability to enable cassandra password authentication

git-subtree-dir: cassandra-local
git-subtree-split: 3c224fdd046c0cab794f47de60f9232e20f83b70

Commit:
Merge commit '4d939e3e9a97baa1e1f20a9aa0b8457510fa9036' into authentication

Commit:
Add SkbKontur prefix to tests namespace.

Commit:
Add authentication ability to ThriftConnection

AndrewKostousov commented on Cassandra.ThriftClient.Tests/Cassandra.ThriftClient.Tests.csproj:
```diff
@@ -3,15 +3,17 @@
   <PropertyGroup>
     <IsPackable>false</IsPackable>
     <TargetFrameworks>net48;netcoreapp2.1;netcoreapp3.1</TargetFrameworks>
+    <RootNamespace>SkbKontur.Cassandra.ThriftClient.Tests</RootNamespace>
```
По мне так не стоило пространство имен в тестах менять. Оно совпадало с именем сборки и ОК.

Предлагаю отревертить эту правку.

iperevoschikov replied:
Старый неймспейс вызывал коллизии с неймспейсом пакета CassandraCSharpDriver (Cassandra.*) там нужно было для ConsistencyLevel-а, Logger-а и нескольких эксепшнов указывать полное имя. Дифф с новым неймспейсом мне показался красивее. Если ты не видишь в этом проблемы, то окей, отменю правку

AndrewKostousov replied:
Ну ок тогда

AndrewKostousov commented on Cassandra.ThriftClient.Tests/FunctionalTests/CustomNodeTests/AuthenticationTest.cs:
```diff
@@ -0,0 +1,105 @@
+                    GossipPort = 8400
+                };
+            node.Restart(TimeSpan.FromMinutes(1));
+            Thread.Sleep(TimeSpan.FromSeconds(30)); // required to initialize default superuser
```
Стоит заменить это на явное ожидание, когда какая-то опреация сможет выполниться (после создания superuser-а) при помощи `Wait(() = {...}, timeout: TimeSpan.FromSeconds(30))`.

AndrewKostousov commented on CHANGELOG.md:
```diff
@@ -1,5 +1,8 @@
 # Changelog
 
+## v2.4.1 - 2020.08.19
```
Тут дату релиза нужно  будет не забыть поменять.

iperevoschikov replied:
Откатил, а то не уверен, какая там будет версия в итоге. Потом прямо в мастер закоммичу.

AndrewKostousov commented on Cassandra.ThriftClient.Tests/FunctionalTests/CustomNodeTests/AuthenticationTest.cs:
```diff
@@ -0,0 +1,105 @@
+                                                                }
+                                                        }
+                                                }
+                                        }, false));
```
Стоит дописать имя параметра. False что?

AndrewKostousov commented on Cassandra.ThriftClient.Tests/FunctionalTests/CustomNodeTests/AuthenticationTest.cs:
```diff
@@ -0,0 +1,105 @@
+        private static void AssertAuthenticationSuccess(ICassandraClusterSettings settings)
+        {
+            var cluster = new CassandraCluster(settings, new SilentLog());
+            var cassandraSchemaActualizer = new CassandraSchemaActualizer(cluster, null, Logger.Instance);
```
Стоит дописать имя параметра. Null что?

AndrewKostousov commented on Cassandra.ThriftClient.Tests/FunctionalTests/CustomNodeTests/AuthenticationTest.cs:
```diff
@@ -0,0 +1,105 @@
+        }
+
+        [Test]
+        public void TestAuthenticatedConnectionNonDefaultCredentials()
```
Стоит еще написать тест (или расширить этот), в котором ассертится, что от имени незареганного пользователя операция не проходит.

AndrewKostousov commented on Cassandra.ThriftClient/Abstractions/Credentials.cs:
```diff
@@ -0,0 +1,14 @@
+            Password = password;
+        }
+
+        public string Username { get; }
```
NotNull и проверка на непустоту в конструкторе

Commit:
Review fixes:

- check credentials argument to be non-empty
- add test for non authenticated connection

AndrewKostousov commented on Cassandra.ThriftClient.Tests/FunctionalTests/CustomNodeTests/AuthenticationTest.cs:
```diff
@@ -0,0 +1,105 @@
+        [Test]
+        public void TestAuthenticatedConnectionNonDefaultCredentials()
+        {
+            var session = GetClusterBuilder().Build()
```
using?

Cluster кстати тоже Disposable

Commit:
Respect IDisposable

Commit:
revert changelog
